In [1]:
import numpy as np
import scipy.linalg as linalg
import math

### 1. Задание

##### Напишите уравнение параболы, проходящей через три точки (x,y): (1,2), (3,10), (5,1)

Парабола задается квадратичной функцией: y = ax^2 + bx + c

Отсюда получим систему уравнений:

a + b + c = 2

9a + 3b + c = 10

25a + 5b + c = 1

Решаем систему уравнений приведенную выше систему уравнений и находим коэффициенты a, b, c:

In [2]:
A = np.array([1,1,1,9,3,1,25,5,1]).reshape(3,3)
B = np.array([2,10,1]).reshape(3,1)
C = np.linalg.solve(A,B)
C

array([[-2.125],
       [12.5  ],
       [-8.375]])

Т.о. уравнение параболы имеет вид: -2.125*x^2 + 12.5*x - 8.375

##### 2. Задание 

Известно, что свежий огурец на 99% состоит из воды. Месяц назад взвесили мешок со свежими огурцами. Получилось, что огурцов ровно 100 кг. Мешок убрали, а через месяц снова взвесили. Огурцы за это время усохли, и теперь вода составляет уже только 98% их веса. Сколько теперь (в кг) весят огурцы?

Другими словами, если бы огурцы сублимировали для использования в "Доширак", то в "сухом" виде их начальный вес составлял бы 1кг (1% от общего веса). Спустя месяц этот "сухой" вес равен 2% от общего веса, но в абсолютных величинах он не изменился и составляет тот же 1 кг. Т.о., общий вес огурцов после усыхания составляет 1кг / 2% = 50кг

#### 3. Задание. Определение логарифма. Решить уравнения:


Логарифм числа b по основанию a определяется как показатель степени, в которую надо возвести основание a, чтобы получить число b. Из определения следует, что нахождение x = log(a)b равносильно решению уравнения a^x=b. Например, log(2)8=3, потому что 2^3=8.

##### 2^x=256

In [3]:
x = math.log(256, 2)
x

8.0

##### 2^x=300

In [4]:
x = math.log(300, 2)
x

8.228818690495881

###### log8(2)^(8x-4)=4

Преобразуем уравнение в следующий вид: 8^4=2^(8x-4), или 2^12=2^(8x-4), т.е. x = 2

###### 3^log9(5x-5)=5

Преобразуем уравнение в следующий вид: 3^y=5, найдем y:

In [5]:
y = math.log(5, 3)
y

1.4649735207179269

Т.о., log9(5x-5)=1.4649735207179269, или 9^1.4649735207179269=5x-5, или 24.999999999999982=5x-5, т.е. x=6

##### x^(log3(x+1))=9

Преобразуем уравнение в вид: log(3)x^(log(3)+1)=log(3)9; отсюда log(3)x^(log(3)+1) = 2; отсюда x равен корню (log(3)+1) из 9; отсюда x = 3

### 4. Задание Свойства логарифмов. Вычислить:

In [11]:
math.log(16, 4)

2.0

In [12]:
math.log(1/25, 5)

-2.0

In [13]:
math.log(5, 25)

0.5

In [14]:
math.log(math.sqrt(27), 3)

1.5

In [15]:
math.log(12, 2) - math.log(3, 2)

2.0

In [16]:
math.log(12, 6) + math.log(3, 6)

2.0

In [17]:
math.e**math.log(5)

4.999999999999999

In [18]:
math.log(225, 2) / math.log(15, 2)

2.0

In [19]:
math.log(32, 4) + math.log(10, 0.1)

1.4999999999999998

In [20]:
9**math.log(math.sqrt(5), 3)

4.999999999999999